spacetemplate/parsed_template.go

```diff
@@ -0,0 +1,239 @@
+var _ convert.Equaler = (*WorkItemTypeHelper)(nil)
+
+// Ensure WorkItemLinkTypeHelper implements the Equaler interface
+var _ convert.Equaler = WorkItemLinkTypeHelper{}
+var _ convert.Equaler = (*WorkItemLinkTypeHelper)(nil)
+
+// Equal returns true if two WorkItemLinkTypeHelper objects are equal; otherwise
+// false is returned.
+func (o WorkItemLinkTypeHelper) Equal(u convert.Equaler) bool {
+	other, ok := u.(WorkItemLinkTypeHelper)
+	if !ok {
+		return false
+	}
+	if o.SourceType != other.SourceType {
+		return false
+	}
+	if o.TargetType != other.TargetType {
+		return false
+	}
+	return o.WorkItemLinkType.Equal(other.WorkItemLinkType)
+}
+
+// Equal returns true if two WorkItemTypeHelper objects are equal; otherwise
+// false is returned.
+func (o WorkItemTypeHelper) Equal(u convert.Equaler) bool {
+	other, ok := u.(WorkItemTypeHelper)
+	if !ok {
+		return false
+	}
+	if o.ExtendsType != other.ExtendsType {
+		return false
+	}
+	return o.WorkItemType.Equal(other.WorkItemType)
+}
+
+// Equal returns true if two ParsedTemplate objects are equal; otherwise
+// false is returned.
+func (pst ParsedTemplate) Equal(u convert.Equaler) bool {
+	other, ok := u.(ParsedTemplate)
+	if !ok {
+		return false
+	}
+	if len(pst.WITs) != len(other.WITs) {
+		return false
+	}
+	if len(pst.WITs) != len(other.WITs) {
+		return false
+	}
+	for k := range pst.WITs {
+		if other.WITs[k] == nil {
+			return false
+		}
+		if !pst.WITs[k].Equal(*other.WITs[k]) {
+			return false
+		}
+	}
+	if len(pst.WILTs) != len(other.WILTs) {
+		return false
+	}
+	for k := range pst.WILTs {
+		if other.WILTs[k] == nil {
+			return false
+		}
+		if !pst.WILTs[k].Equal(*other.WILTs[k]) {
+			return false
+		}
+	}
+	return true
+}
+
+// String convert a parsed template into a string in YAML format
+func (pst ParsedTemplate) String() string {
+	bytes, err := yaml.Marshal(pst)
+	if err != nil {
+		log.Info(nil, map[string]interface{}{
+			"err": err,
+		}, "failed to marshal space template to YAML")
+		return ""
+	}
+	return string(bytes)
+}
+
+// ParseStringToTemplate parses a string into a parsed template object
+func ParseStringToTemplate(templ string) (*ParsedTemplate, error) {
+	var parsedTempl ParsedTemplate
+	if err := yaml.Unmarshal([]byte(templ), &parsedTempl); err != nil {
+		log.Info(nil, map[string]interface{}{
+			"template": templ,
+			"err":      err,
+		}, "failed to unmarshal YAML space template")
+		return nil, errors.NewBadParameterError("template", templ).Expected("YAML space template")
+	}
+	return &parsedTempl, nil
+}
+
+// getIDForPredefinedWIT returns the ID if the given WIT name/ID can be assumed
+// to exist even though it wasn't specified by the space template itself
+func getIDForPredefinedWIT(wit string) *uuid.UUID {
+	if wit == "" {
+		return nil
+	}
+	preExistingWITs := map[string]uuid.UUID{
+		"planneritem": workitem.SystemPlannerItem,
+	}
+	for refName, id := range preExistingWITs {
+		if refName == wit || id.String() == wit {
+			return &id
```

Comment:
I just says that if you somewhere reference either `planneritem` or `86af5178-9b41-469b-9096-57e5155c3f31` in the space template, it will return `86af5178-9b41-469b-9096-57e5155c3f31` for reference. Currently the logic might seem odd because it is only one pre-defined work item type.